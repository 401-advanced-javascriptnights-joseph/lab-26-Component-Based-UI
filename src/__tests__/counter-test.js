import React from 'react';
import renderer from 'react-test-renderer';
import Counter from '../components/counter/counter.js';

describe('<Counter />', () => {
  it('is alive at start', () => {
    let component = shallow(<Counter />);
    expect(component.find('span').exists()).toBeTruthy();
  });

  it('counter decrements when minus sign is clicked', () => {
    let component = mount(<Counter />);
    let button = component.find('#decrement');
    button.simulate('click');
    expect(component.state('count')).toBe(-1);
  });

  it('counter increment when plus sign is clicked', () => {
    let component = mount(<Counter />);
    let button = component.find('#increment');
    button.simulate('click');
    expect(component.state('count')).toBe(+1);
  });

  it('counter decrements when minus sign is clicked', () => {
    let component = mount(<Counter />);
    let displayElement = component.find('span');
    expect(displayElement.text()).toContain(`Count:`)
  });

  it('counter increment when plus sign is clicked', () => {
    let component = mount(<Counter />);
    let displayElement = component.find('span');
    expect(displayElement.text()).toContain('Count:')
  });

  it('renders correctly', () => {
    const tree = renderer.create(<Counter />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});