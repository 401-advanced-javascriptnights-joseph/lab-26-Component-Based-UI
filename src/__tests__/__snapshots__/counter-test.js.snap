// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Counter /> render correctly 1`] = `
<div
  id="counter"
>
  <button
    id="decrement"
    onClick={[Function]}
  >
    -
  </button>
  <span
    style={
      Object {
        "background": "yellow",
      }
    }
  >
    Count: 
    0
     
  </span>
  <button
    id="increment"
    onClick={[Function]}
  >
    +
  </button>
</div>
`;

exports[`<Counter /> renders correctly 1`] = `
<div
  id="counter"
>
  <button
    id="decrement"
    onClick={[Function]}
  >
    -
  </button>
  <span
    style={
      Object {
        "background": "yellow",
      }
    }
  >
    Count: 
    0
     
  </span>
  <button
    id="increment"
    onClick={[Function]}
  >
    +
  </button>
</div>
`;
